### DISCUSSION TITLE
Trying to implement a demo app

### DISCUSSION THREAD

**Discussion by mward-sudo** [liveshowy/webauthn_components#28] [General]
Thanks for the project and I look forward to watching it develop over time.

I'm trying to implement a demonstration app for my personal website using `webauthn_live_component` with `Phoenix 1.7rc.0` and `LiveView 0.18`, but I'm having some problems getting to work as expected.

I understand that the code is still early alpha, and as such I expect things to be in flux and for problems to occur.

Currently I am not quite sure why my demo app isn't working. If someone can help me to get it to work then I would be more than happy to write detailed documentation for the project on how to get a simple demo app up and running.

My repo is https://github.com/mward-sudo/michaelward_uk/ and the branch for these changes is `issue-1`. I am using the `23-update_deps` branch of `webauthn_live_component` as it has changes that I require.

My intention for this demo is to have a `LiveView` page at `/auth` that allows for registration of a new user and logging in for an existing user. If the user is signed in then the page should not show the login form, but a message saying that user is logged in and offering a sign out option.

Registration of a new user works and the user is logged in after registering, and is able to sign out.

However, using the login functionality for an existing user does not work correctly. 

Any help with this would be appreciated as I am currently blind as to what I am doing wrong.

It would also be good if the user login persisted between sessions / page reloads and was visible to non LiveView sections of the app, but these are secondary concerns.

**Comment by type1fool** [comment]
Yikes! Sorry @mward-sudo that I didn't see this until now. I'll see if I can get notifications enabled for discussions on this repo.

Copying an update from another discussion:

> I have been working on some big improvements to the project over the last few weeks, primarily token improvements and code generators. [PR 50](https://github.com/liveshowy/webauthn_components/pull/50) is open, and I'm revising documentation to capture the changes. I would appreciate feedback on the PR if you are so inclined. Getting started should be much easier once this work is completed.

Thank you.
